server/project-service/src/main/java/edu/illinois/ncsa/incore/service/project/controllers/ProjectController.java

```diff
@@ -120,12 +121,41 @@ public List<Project> getProjectsList(@Parameter(name = "Name filter") @QueryPara
             queryMap.put("region", region);
         }
 
-        List<Project> projects;
+        List<Project> projects = new ArrayList<>();
 
-        if (queryMap.isEmpty()) {
+        if (searchText != null && !searchText.isEmpty()) {
+            Project project = this.projectDAO.getProjectById(searchText);
```

Comment:
Just curious. Is there any occasion that the searchText is null but searchText is not empty or the other way around?

Reply:
oh the other way, sometimes text could be empty string ""